pkg/operator/controller/controller_dns_daemonset_test.go

```diff
@@ -469,14 +481,23 @@ func TestDaemonsetConfigChanged(t *testing.T) {
 			expect: true,
 		},
 		{
-			description: "if the update strategy changes",
+			description: "if the update strategy's max unavailable parameter changes",
 			mutate: func(daemonset *appsv1.DaemonSet) {
-				daemonset.Spec.UpdateStrategy = appsv1.DaemonSetUpdateStrategy{
-					Type: appsv1.RollingUpdateDaemonSetStrategyType,
-					RollingUpdate: &appsv1.RollingUpdateDaemonSet{
-						MaxUnavailable: pointerTo(intstr.FromString("10%")),
-					},
-				}
+				daemonset.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable = pointerTo(intstr.FromString("10%"))
+			},
+			expect: true,
+		},
+		{
+			description: "if the update strategy's max surge parameter changes",
+			mutate: func(daemonset *appsv1.DaemonSet) {
+				daemonset.Spec.UpdateStrategy.RollingUpdate.MaxSurge = pointerTo(intstr.FromString("10%"))
+			},
+			expect: true,
+		},
+		{
+			description: "if spec.minReadySeconds changes",
+			mutate: func(daemonset *appsv1.DaemonSet) {
+				daemonset.Spec.MinReadySeconds = 9
```

Comment:
```suggestion
				daemonset.Spec.MinReadySeconds = pointerTo(intstr.FromInt(9))
```

Reply:
`minReadySeconds` takes an integer value: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#daemonsetspec-v1-apps https://github.com/kubernetes/api/blob/ec59454f24f514bb472628bc0a5c8c9125218dbc/apps/v1/types.go#L391-L395

Reply:
Sorry, I confused it with MaxUnavailable.